### DISCUSSION TITLE
Source Generators for MauiIcons Custom Font Icons

### DISCUSSION THREAD

**Discussion by AathifMahir** [AathifMahir/MauiIcons#29] [Ideas]
Hello Folks,

I was looking to develop a source generation core for **MauiIcons** to accustom .ttf, .tof and .wof. This brings in the ability to provide your own font files within the above extensions and the library would automatically generate the icons based on the provided font file for intellisense suggestions and etc..

As far as now, this is an initial idea and real Implementation would require some expertise with source generation and contribution from others.

**Initial problems:**
- We need to develop or find any Open source libraries to encode and decode .ttf, .tof, and .wof font files using c#
- Sort out how to architect the source generation and the performance aspect of generating large font files and etc..
- Other Miscellaneous Stuffs

I would love to hear from you all and suggestions on this and anything that efficiently fills this purpose.

**Comment by HavenDV** [comment]
Maybe this will help you. I use this generator for easy access to Embedded resources in tests - https://github.com/HavenDV/H.Resources.Generator

**Reply by AathifMahir** [reply]
Thanks for sharing this, I'll look into the possibilities of using it on MauiIcons
